src/pc/sdl2.c

```diff
@@ -236,19 +332,63 @@ DISPENV* MyPutDispEnv(DISPENV* env) {
 }
 
 void MySetDrawMode(DR_MODE* p, int dfe, int dtd, int tpage, RECT* tw) {
-    if (tpage >= LEN(g_SdlVramTextures)) {
-        DEBUGF("tpage %d, how to handle?", tpage);
-        return false;
-    }
-    if (!g_SdlVramTextures[tpage] && !InitializeTexture(tpage)) {
-        return;
+    g_Tpage = tpage;
+}
+
+SDL_Texture* GetVramTexture(int tpage, int clut) {
+    if (g_LastVramTexTpage != tpage) {
+        g_LastVramTexTpage = tpage;
+
+        u8* src = g_RawVram;
+        src += (tpage & 0xF) * 128;
+        src += ((tpage >> 4) & 1) * VRAM_STRIDE * 256;
+
+        u16* pal = g_RawVram;
+        pal += clut * 0x10;
+        // Hack that rewrites the VRAM to adjust transparent palette colors
+        pal[0] &= ~0x8000; // first color always transparent?
```

Comment:
According to this 0x0000 is transparent. What's the issue you're seeing?
<img width="714" alt="image" src="https://github.com/Xeeynamo/sotn-decomp/assets/122322823/55f7382e-e5a7-42a7-a079-db92e12dd18f">